modules/leveling/cards.py

```diff
@@ -70,43 +70,73 @@ async def generate_rank_card(user, level_data):
     buffer.seek(0)
     return discord.File(buffer, filename="shadow_rank_card.png")
 
-async def generate_leaderboard_card(self, top_users):
-    width, height = 800, 70 * len(top_users) + 80
-    image = Image.new("RGBA", (width, height), (13, 13, 13, 255))
-    draw = ImageDraw.Draw(image)
-
-    font_title = ImageFont.truetype(font_path, 28)
-    font_entry = ImageFont.truetype(font_path, 20)
+async def generate_leaderboard(bot: discord.Bot, users, start_index=0, current_user_id=None, page=1, total_pages=1):
```

Comment:
You should type at least the users variable as well as the return type properly so an outsider can see what's going on, as for users, it's a bit unclear from the code